packages/xstate-fsm/src/types.ts

```diff
@@ -96,6 +96,9 @@ export namespace StateMachine {
   > {
     config: StateMachine.Config<TContext, TEvent>;
     initialState: State<TContext, TEvent, TState>;
+    getStateByValue: (
```

Comment:
does this have to be a public method?

Reply:
I don't see a better approach. If you have an idea how to avoid having a private method, I'm ready to change the implementation.

The current implementation of the Service keeps track of the current state by keeping the full `State` object from the state machine. It works fine if the state machine always starts from the initial state: the state machine can expose initial state directly. 

This approach does not work, If we consider that a service can be started in any state. We need a way to receive or derive the `State` object from the state value (another approach is to ask the developer to provide the full object). It is reasonable to have a method of machine that returns a state object because the state is created from the machine configuration and a list of actions provided.

Reply:
Note how in the core XState we do not "resolve" entry actions when rehydrating state: https://codesandbox.io/s/competent-chaum-kf9qb?file=/src/index.js

Ideally, we would match its behavior here to avoid confusion. And to do that you would only have to recreate the State using publicly known information: value & context.

Reply:
@Andarist take a look. I rolled back things to the previous state and manually create the state object. 

The entry action of the rehydrated state will bot be triggered

Reply:
It looks good. I have only one concern - it seems to me that rehydrating context is very important for actual state persistence and this is not covered by this PR.

Reply:
I totally agree, this is another part that is not handled now. 

I'd love to implement it in a separate PR. I've checked the main library and there are a bunch of options available. We can go an easy way and allow a user to provide context as a second argument. This should not increase the size of the lib.

Another option is to create `withContext` method of the `StateMachine` that sets the context. ([Reference to the main library](https://xstate.js.org/docs/guides/context.html#initial-context))

What do you think?

Reply:
Wouldnt something like this work: https://github.com/nlopin/xstate/pull/1 ?

Reply:
Merged. I'll add the test today and we are good to go 👍

Reply:
I added the test and grouped all `start`-related tests